perf-tool/src/monitor.cpp

```diff
@@ -53,6 +56,18 @@ JNIEXPORT void JNICALL MonitorContendedEntered(jvmtiEnv *jvmtiEnv, JNIEnv *env,
 	    sprintf(str, "0x%x", hash);
 	    j["monitorHash"] = str;
     }
+   
+    {
+    std::lock_guard<std::mutex> lg(monitorMapMutex);	    
+    if(m_monitor.find(hash) == m_monitor.end())
+        m_monitor[hash] = 1;
+    else
+        {
+         int val = m_monitor[hash];
+         m_monitor[hash] = ++val;
+         j["per_object_contended_count"] = val;
+        }
+    }
```

Comment:
This sequence is inefficient because it accesses the hashtable 3 times: `find' and two `[ ]` operations.
We could use the same pattern I used for the monitors:
```
                    std::lock_guard<std::mutex> lg(contentionMutex);
                    samplesPerObjectType = ++(numContentions[std::string(str)]);
```

Reply:
Also, `j["per_object_contended_count"] = val;` does not have to be protected by a monitor.